### DISCUSSION TITLE
Location API

### DISCUSSION THREAD

**Discussion by vshcherb** [osmandapp/OsmAnd#11202] [Q&A] [answered]
Hello! Please, tell me, does the osmand API support the following option: get the coordinates of the user's current position and/or the coordinates of the point specified by the user on the map in the calling application? Thank!

**Comment by vshcherb** [comment, marked as answer]
All API is part of AIDL file - https://github.com/osmandapp/OsmAnd/blob/master/OsmAnd-api/src/net/osmand/aidlapi/IOsmAndAidlInterface.aidl.  I think this is the method - https://github.com/osmandapp/OsmAnd/blob/master/OsmAnd/src/net/osmand/aidl/OsmandAidlApi.java#L1801.

**Reply by dimio** [reply]
Many thanks!

**Comment by richardeschloss** [comment]
Hi @vshcherb , I really like the work you and the team have done on Osmand. Are there documentation on how to use the API with simple HTTP requests? For example, to generate a nautical route, I know I can use the following HTTP request:

> https://maptile.osmand.net/routing/route?routeMode=boat,height=0,width=0,nativerouting,nativeapproximation,calcmode=&points=48.522285,-4.042969&points=51.730620,-8.349609&maxDist=30

A few questions: 
1. Is there a place where this API is documented? I think something like [these API docs](https://marinas.com/developers/api_documentation#point-retrieve) would be extremely helpful for developers because they can quickly try things out in the browser. Is Osmand passing our API requests to an external service like OpenStreetmap? I'm particularly interested in nautical routing.
2. I understand the routing API is limited to 30 km. Is there any way we would be able to expand our search beyond that? A paid tier?
